worlds/stardew_valley/bundles/bundle.py

```diff
@@ -134,30 +125,55 @@ def can_appear(self, options: StardewValleyOptions) -> bool:
+    def __init__(self, room: str, name: str, categories: List[List[BundleItem]], number_possible_items: int,
+                 number_required_items: int):
         super().__init__(room, name, [], number_possible_items, number_required_items)
         self.categories = categories
 
-    def create_bundle(self, bundle_price_option: BundlePrice, random: Random, options: StardewValleyOptions) -> Bundle:
-        if bundle_price_option == BundlePrice.option_minimum:
-            number_required = 1
-        elif bundle_price_option == BundlePrice.option_maximum:
-            number_required = 8
-        else:
-            number_required = self.number_required_items + bundle_price_option.value
+    def create_bundle(self, bundle_price_option: BundlePrice, random: Random, content: StardewContent,
+                      options: StardewValleyOptions) -> Bundle:
+        number_required, price_multiplier = get_bundle_final_prices(bundle_price_option, self.number_required_items, False)
         number_categories = len(self.categories)
         number_chosen_categories = self.number_possible_items
         if number_chosen_categories < number_required:
             number_chosen_categories = number_required
 
         if number_chosen_categories > number_categories:
-            chosen_categories = self.categories + random.choices(self.categories, k=number_chosen_categories - number_categories)
+            chosen_categories = self.categories + random.choices(self.categories,
+                                                                 k=number_chosen_categories - number_categories)
         else:
             chosen_categories = random.sample(self.categories, number_chosen_categories)
 
         chosen_items = []
         for category in chosen_categories:
-            filtered_items = [item for item in category if item.can_appear(options)]
+            filtered_items = [item for item in category if item.can_appear(content, options)]
             chosen_items.append(random.choice(filtered_items))
 
+        chosen_items = [item.as_amount(max(1, math.floor(item.amount * price_multiplier))) for item in chosen_items]
         return Bundle(self.room, self.name, chosen_items, number_required)
+
+
+def get_bundle_final_prices(bundle_price_option: BundlePrice, default_required_items: int, is_currency: bool) -> Tuple[int, float]:
+    number_required_items = get_number_required_items(bundle_price_option, default_required_items)
+    price_multiplier = get_price_multiplier(bundle_price_option, is_currency)
+    return number_required_items, price_multiplier
+
+
+def get_number_required_items(bundle_price_option: BundlePrice, default_required_items: int) -> int:
+    if bundle_price_option == BundlePrice.option_minimum:
+        return 1
+    if bundle_price_option == BundlePrice.option_maximum:
+        return 8
+    number_required = default_required_items + bundle_price_option.value
+    return max(1, number_required)

```

Comment:
Are there bundles with more than 9 items required by default? If not, the first if statement can probably be removed, but doesn't necessarily need to be since that makes it more obvious what it's doing.

Are there bundles with 7 or more items required by default? If so, you'll return 9 or more with the very_expensive option for those bundles, making them higher than the maximum option, which does not sound intended.

This could probably be reduced to the below block, if you want to ensure bundle prices are never lower than 1 or higher than 8.
```py
def get_number_required_items(bundle_price_option: BundlePrice, default_required_items: int) -> int:
    number_required = default_required_items + bundle_price_option.value
    return min(8, max(1, number_required))
```

Reply:
While I do understand and agree that this code would work, I prefer the above version because it is more explicit about what it does.
Technically, the game UI would allow up to 12 items, but I decided that maximum was 8 because that happened to match what was the biggest bundle on very expensive (the biggest one is 6), so it felt like a good milestone to decide "this is the max, now I'll apply to everything".

But you did catch a potential issue, as I am making custom bundles willy-nilly, and someday I might make one with a high enough default price that it goes over 8 on the expensive options. So I'll take the min(max(value)) part of the sugestion